src/main/scala/Column.scala

```diff
@@ -1134,6 +1161,90 @@ case class Column private (
     }
   }
 
+  def validateValue(value: Any): Array[ErrorWithoutContext] =
+    value match {
+      case numericValue: Number => {
+        validateNumericValue(numericValue)
+      }
+      case s: String => Array[ErrorWithoutContext]()
```

Comment:
When does the `String` case happen? I thought this only got called for numeric/datetime/range datatypes?

Reply:
No, validateValue is called for every successfully parsed column value. Which can be strings, booleans, numbers, dates isn't it? I have introduced the datetime and boolean as well now.

Reply:
Or else we could use a conditional check and invoke this function only for `numeric/datetime/range` datatypes.

Reply:
No, I think this is probably find, no point doing two match statements where one would do.